### DISCUSSION TITLE
Material Symbols Implementation

### DISCUSSION THREAD

**Discussion by AathifMahir** [AathifMahir/MauiIcons#53] [General]
# Material Symbols

Material Symbols is Latest Version of Google's Material Font, The Font is Designed Based on Variable Font Specs and It's in Use with Material 3 Design Language to Complement Each other

# Problem

Since Material Symbols Font is Variable Font, A Single Unicode Does Have over 100+ Different Variants and As Whole Package that Contains All the Variants and Fonts would Allocates Around 10MB in DLL Size, Therefore We need to Get Rid of Variants that We Don't Need.

- First, we need to Look Through Optical Size Variants, As Far as Optical Size 24px is What Most Font Comes as Default, As Far as Considering the Advantages of Variable Font Scaling Specs, we could go with 48px Variant and That Would be High Quality and Clarity Side.

> **This Reduces Variants from 100+ to 42**

- Second, we need to Get Rid of Weight from 7 Different Variants to about 3 or 4 Variants, Else At Least 5 Weight Variants.

> **This Reduces Variants from 42 to 30**

### More Info Follow-up Soon as We Move Ahead with the Plan

**Comment by vhugogarcia** [comment]
I do agree with the optical size, let's try 48px. 

About the weight, .NET MAUI does not implement this feature within their API. To achieve this you need to implement it via handlers. 

I think weight can be a feature for future releases. We can release the default one built in.
